src/Request/ReportRequest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare(strict_types=1);
+
+namespace Paytrail\SDK\Request;
+
+use Paytrail\SDK\Exception\ValidationException;
+use Paytrail\SDK\Util\ObjectPropertyConverter;
+
+
+class ReportRequest implements \JsonSerializable
+{
+    const PAYMENT_STATUSES = [
+        'default',
+        'paid',
+        'all',
+        'settled'
+    ];
+
+    private $requestType;
+    private $callbackUrl;
+    private $paymentStatus;
+    private $startDate;
+    private $endDate;
+    private $limit;
+    private $reportFields;
+    private $subMerchant;
+
+    use ObjectPropertyConverter;
+
+    public function validate()
```

Comment:
Optional formatting comment:
Nested if statements inside validation could be replaced with:
`isset($props[<property_name>]) && <property_validation>` format to avoid nested if statements.

Reply:
Fixed